photon-test/src/main/scala/com/linkedin/photon/ml/test/TestTemplateWithTmpDir.scala

```diff
@@ -24,15 +27,25 @@ import org.testng.annotations.AfterMethod
  * @author dpeng
  */
 trait TestTemplateWithTmpDir {
+
+  private val tmpDirs = new mutable.ArrayBuffer[String]()
+
   /**
    * Return the temporary directory as a string.
    * @return the temporary directory as a string.
    */
-  def getTmpDir: String = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+  def getTmpDir: String = {
+    val tmpDirName = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+    tmpDirs.synchronized {
+      tmpDirs += tmpDirName
+    }
+    tmpDirName
+  }
 
   @AfterMethod
```

Comment:
Yeah. This could potentially delete another thread's temporary folder before that thread is done with its tests.

It's strange that ```FileUtils.forceDeleteOnExit``` isn't working here. Any idea why?